src/dune_rules/utop.ml

```diff
@@ -38,9 +38,7 @@ let libs_and_ppx_under_dir sctx ~db ~dir =
   | None -> Memo.return ([], [])
   | Some dir ->
     let+ libs, pps =
-      Source_tree_map_reduce.map_reduce
-        dir (* TODO this is wrong under [(subdir ..)] *)
-        ~traverse:{ data_only = false; vendored = true; normal = true }
+      Source_tree_map_reduce.map_reduce dir ~traverse:Sub_dirs.Status.Set.all
```

Comment:
This can be made more efficient if anyone cares. We don't need to scan all `data_only` dirs. Only the ones where the parent declared `(subdir ..)`